pkg/apis/provisioning/v1alpha5/requirements.go

```diff
@@ -70,131 +75,228 @@ var (
 }
 
-func (r Requirements) Add(requirements ...v1.NodeSelectorRequirement) Requirements {
-	return append(r, Requirements(requirements).Normalize()...)
-}
-
-// Normalize the requirements to use WellKnownLabels
-func (r Requirements) Normalize() Requirements {
-	normalized := Requirements{}
-	for _, requirement := range r {
-		label := requirement.Key
-		if normalized, ok := NormalizedLabels[requirement.Key]; ok {
-			label = normalized
-		}
-		normalized = append(normalized, v1.NodeSelectorRequirement{Key: label, Operator: requirement.Operator, Values: requirement.Values})
-	}
-	return normalized
-}
-
-// Consolidate combines In and NotIn requirements for each unique key, producing
-// an equivalent minimal representation of the requirements. This is useful as
-// requirements may be appended from a variety of sources and then consolidated.
-// Caution: If a key has contains a `NotIn` operator without a corresponding
-// `In` operator, the requirement will permanently be [] after consolidation. To
-// avoid this, include the broadest `In` requirements before consolidating.
-func (r Requirements) Consolidate() (requirements Requirements) {
-	for _, key := range r.Keys() {
-		requirements = requirements.Add(v1.NodeSelectorRequirement{
-			Key:      key,
-			Operator: v1.NodeSelectorOpIn,
-			Values:   r.Requirement(key).UnsortedList(),
-		})
-	}
-	return requirements
+// NewRequirements constructs requiremnets from NodeSelectorRequirements
+func NewRequirements(requirements ...v1.NodeSelectorRequirement) Requirements {
+	return Requirements{requirements: map[string]sets.Set{}}.Add(requirements...)
 }
 
-func LabelRequirements(labels map[string]string) (r Requirements) {
+// NewLabelRequirements constructs requriements from labels
+func NewLabelRequirements(labels map[string]string) Requirements {
+	requirements := []v1.NodeSelectorRequirement{}
 	for key, value := range labels {
-		r = r.Add(v1.NodeSelectorRequirement{Key: key, Operator: v1.NodeSelectorOpIn, Values: []string{value}})
+		requirements = append(requirements, v1.NodeSelectorRequirement{Key: key, Operator: v1.NodeSelectorOpIn, Values: []string{value}})
 	}
-	return r
+	return NewRequirements(requirements...)
 }
 
-func PodRequirements(pod *v1.Pod) (r Requirements) {
+// NewPodRequirements constructs requirements from a pod
+func NewPodRequirements(pod *v1.Pod) Requirements {
+	requirements := []v1.NodeSelectorRequirement{}
 	for key, value := range pod.Spec.NodeSelector {
-		r = r.Add(v1.NodeSelectorRequirement{Key: key, Operator: v1.NodeSelectorOpIn, Values: []string{value}})
+		requirements = append(requirements, v1.NodeSelectorRequirement{Key: key, Operator: v1.NodeSelectorOpIn, Values: []string{value}})
 	}
 	if pod.Spec.Affinity == nil || pod.Spec.Affinity.NodeAffinity == nil {
-		return r
+		return NewRequirements(requirements...)
 	}
+	// The legal operators for pod affinity and anti-affinity are In, NotIn, Exists, DoesNotExist.
```

Comment:
This comment is inaccurate. This is not pod affinity. This is node affinity. There is no node antiaffinity.  I think you can get away with not having any comment at all.
```suggestion
```